src/cargo/core/profiles.rs

```diff
@@ -330,6 +330,7 @@ impl Profiles {
         result.root = for_unit_profile.root;
         result.debuginfo = for_unit_profile.debuginfo;
         result.opt_level = for_unit_profile.opt_level;
+        result.debug_assertions = for_unit_profile.debug_assertions;
```

Comment:
```rust
    /// The profile for *running* a `build.rs` script is only used for setting
    /// a few environment variables. To ensure proper de-duplication of the
    /// running `Unit`, this uses a stripped-down profile (so that unrelated
    /// profile flags don't cause `build.rs` to needlessly run multiple
    /// times).
```

Considering we already have `debuginfo` and `opt_level`, this seems like it shouldn't be an issue for de-duplication of the running `Unit`